Update kernel version to 5.2.0-m4

daneshk opened the pull request:
Resolves: #29 , #32

lmanohara committed:
Update kernel version to 5.2.0-m3

lmanohara committed:
Incorporating review comments

daneshk committed:
Get datasource-core dependency version from carbon.datasource.version

daneshk committed:
Add config provider support for carbon-deployment

Enable global configuration model support in carbon-deployment. Remove read database configuration from *-datasource.xml, instead read configurations from deployment.yaml.

Resolves #29

daneshk committed:
review changes: change the license year to 2016 in existing files

vidurananayakkara commented on components/org.wso2.carbon.datasource.core/src/main/java/org/wso2/carbon/datasource/core/beans/DataSourceDefinition.java:
```diff
@@ -15,24 +15,41 @@
+
+    public DataSourceDefinition() {
+        Map<String, String> configMap = new LinkedHashMap<String, String>();
+        configMap.put("jdbcUrl", "jdbc:h2:./database/WSO2_CARBON_DB;DB_CLOSE_ON_EXIT=FALSE;LOCK_TIMEOUT=60000");
```
Should we have the keys as constants in here?

daneshk replied:
This default values are for the configure documentation. we don't use these keys anyway else. only in this class. do we need to specify keys as constants.

vidurananayakkara replied:
If so we can keep as it is (no need to specify keys as constants)

vidurananayakkara commented on components/org.wso2.carbon.datasource.core/src/main/java/org/wso2/carbon/datasource/core/beans/DataSourceMetadata.java:
```diff
@@ -16,25 +16,25 @@
  */
-@XmlRootElement(name = "datasource")
-@XmlType(propOrder = {"name", "description", "jndiConfig", "definition"})
+@Configuration(description = "Data Source metadata configuration")
```
The hashCode method in this class simply returns -1. IMO this method should return an integer-specific to the object instance. (http://stackoverflow.com/questions/3563847/what-is-the-use-of-hashcode-in-java) Is there a specific reason for returning -1 in this method

daneshk replied:
corrected. thanks.

vidurananayakkara commented on components/org.wso2.carbon.datasource.core/src/main/java/org/wso2/carbon/datasource/core/beans/JNDIConfig.java:
```diff
@@ -115,23 +112,25 @@ public boolean equals(Object rhs) {
     @Override
     public int hashCode() {
-        assert false : "hashCode() not implemented";
+//        assert false : "hashCode() not implemented";
```
As mentioned in the previous comment, is there a specific reason for returning -1. Furthermore having commented code is considered as a code smell and therefore IMO we should remove this line. WDYT?

daneshk replied:
corrected. thanks.

vidurananayakkara commented on components/org.wso2.carbon.datasource.core/src/test/java/org/wso2/carbon/datasource/core/BaseTest.java:
```diff
@@ -25,26 +35,44 @@
-        Path configFilePath = Paths.get("src", "test", "resources", "conf", "datasources", "master-datasources.xml");
-        Path configPathCopyLocation = Paths.get("target", "carbonHome", "conf", "datasources",
-                "master-datasources.xml");
+        Path configFilePath = Paths.get("src", "test", "resources", "conf", "deployment.yaml");
```
Should we have deployment.yaml file name as a constant in this class (Since the same file name is used in multiple places)

daneshk replied:
corrected. thanks

vidurananayakkara commented on tests/test-distribution/pom.xml:
```diff
@@ -0,0 +1,177 @@
+            <plugin>
+                <groupId>org.wso2.carbon.maven</groupId>
+                <artifactId>carbon-feature-plugin</artifactId>
+                <version>${carbon.feature.plugin.version}</version>
```
Shouldn't this plugin be specified in the root pom within the `<build><pluginmanagement><plugins>` tag and then be used here? Then we do not need to specify the version of the plugin here and would be common throughout the project.

daneshk committed:
make changes for the review comments, implement hashcode instead sending -1

daneshk committed:
Fix issues and suggestions raised in the code review.

1. remove snakeyaml usage in converting configuration map to config bean.
2. remove hardcoded default values from config bean classes.
3. create configuration yaml file manually and copy to feature.

daneshk committed:
remove snakeyaml dependency from datasource

Niranjan-K commented:
@daneshk : Can we have Git issues for the new commits please? Currently you have put the lengthy commit titles. It would have been better if we have lengthy description but in the commit body.

daneshk committed:
update kernel version to 5.2.0-m4

daneshk committed:
remove snakeyaml dependency

daneshk committed:
Edit non-osgi sample with new config provider

daneshk committed:
Remove feature plugin version in distribution pom

imesh commented on components/org.wso2.carbon.datasource.core/src/main/java/org/wso2/carbon/datasource/core/DataSourceManager.java:
```diff
@@ -127,48 +126,42 @@ public void initDataSources(String configurationDir, Map<String, DataSourceReade
+                dataSourceConfiguration = configProvider.getConfigurationObject(DataSourcesConfiguration.class);
+                if (dataSourceConfiguration.getDataSources() == null && dataSourceConfiguration.getDataSources()
+                    .isEmpty()) {
+                    throw new DataSourceException("configuration doesn't specify any datasource configurations");
```
Would it be more meaningful to make the first letter of the word "configuration" uppercase?

imesh commented on components/org.wso2.carbon.datasource.core/src/main/java/org/wso2/carbon/datasource/core/DataSourceManager.java:
```diff
@@ -127,48 +126,42 @@ public void initDataSources(String configurationDir, Map<String, DataSourceReade
-                dataSourceRepository.addDataSource(carbonDataSource);
-                DataSourceJndiManager.register(carbonDataSource, dataSourceReader);
+                if (logger.isDebugEnabled()) {
+                    logger.debug("Creating datasource object| datasource name: " + dsmInfo.getName());
```
What would be the reason to use the pipe character "object| datasource name"?

daneshk replied:
No specific reason. just wanted to separate the message.

Niranjan-K commented on components/org.wso2.carbon.datasource.core/pom.xml:
```diff
@@ -68,6 +68,11 @@
+        <dependency>
+            <groupId>org.wso2.carbon.config</groupId>
+            <artifactId>org.wso2.carbon.config</artifactId>
+            <optional>true</optional>
```
If we are using this in the java code (in the default flow), then we shouldn't have this dependency as optional.

Niranjan-K commented on components/org.wso2.carbon.datasource.core/pom.xml:
```diff
@@ -110,10 +115,16 @@
+            org.wso2.carbon.datasource.utils.*
+        </export.package>
+        <import.package>
+            javax.xml.parsers;version=0.0.0,
```
We need to change this to a range, say 0.0.0 to 1.0.0

Niranjan-K commented on components/org.wso2.carbon.datasource.core/src/main/java/org/wso2/carbon/datasource/rdbms/hikari/utils/HikariDataSourceUtils.java:
```diff
@@ -16,47 +16,91 @@
-                config.setDataSourceProperties(properties);
+    public static HikariConfig buildConfiguration(Object configuration) throws DataSourceException {
+        if (configuration == null) {
+            throw new DataSourceException("datasource configurations are not provided. datasource object will not be " +
```
Should start with a capital letter.

Niranjan-K commented on components/org.wso2.carbon.datasource.core/src/main/java/org/wso2/carbon/datasource/rdbms/hikari/utils/HikariDataSourceUtils.java:
```diff
@@ -16,47 +16,91 @@
+                    Field beanField = hikariClass.getDeclaredField(name);
+                    if (!beanField.isAccessible()) {
+                        AccessController
+                            .doPrivileged((PrivilegedAction<Object>) () -> {
```
we can move this to the same level as "AccessController"

daneshk committed:
Fix code review suggestions

daneshk committed:
Fix checkstyle issues in windows
